crates/ty_python_semantic/src/types/function.rs

```diff
@@ -690,12 +677,10 @@ impl<'db> FunctionLiteral<'db> {
 #[derive(PartialOrd, Ord)]
 pub struct FunctionType<'db> {
     pub(crate) literal: FunctionLiteral<'db>,
-
-    /// Type mappings that should be applied to the function's parameter and return types. This
-    /// might include specializations of enclosing generic contexts (e.g. for non-generic methods
-    /// of a specialized generic class).
-    #[returns(deref)]
-    type_mappings: Box<[TypeMapping<'db, 'db>]>,
+    #[returns(as_ref)]
+    pub(crate) updated_signature: Option<CallableSignature<'db>>,
+    #[returns(as_ref)]
+    pub(crate) updated_last_definition_signature: Option<Signature<'db>>,
```

Comment:
Maybe some doc comments here explaining what these fields are and when/how they become populated?

Reply:
I added some doc comments and made those fields non-public. @dcreager please feel free to change them in a follow-up, if something seems wrong/missing. I'd like to merge this PR to fix https://github.com/astral-sh/ty/issues/1261.